TMEDIA 280 replace corecomponents package, error handling story feed tag

JackHowa opened the pull request:
## Description
Copy-paste corecomponents package
Add error handling for falsy tag slug

## Jira Ticket
- [TMEDIA-280](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-280)

## Acceptance Criteria
The Story Feed Tags Content Source block no longer has arc-core-components as a dependency

NOTE: Make sure it’s also removed from the block’s package.json!

If no tag is provided, the content source does not make a request to the content api.
## Test Steps

1. Checkout this branch `git checkout TMEDIA-280-story-feed-tag-perf`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/story-feed-tag-content-source-block `
3. Use content debugger and see preexisting request still working like "sports" tag
<img width="724" alt="Screen Shot 2021-07-26 at 09 46 41" src="https://user-images.githubusercontent.com/5950956/127008976-9dc15684-ae27-4a2b-8009-88292edcce3b.png">
4. Throw error if falsy param for tag slug 

## Dependencies or Side Effects
- removing deps 

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

Commit:
perf: Use story-feed tag inline rather than import via package

Commit:
feat: Throw error if tag slug not present

Commit:
test: Add error on falsy tag sections and base tests

via https://github.com/wapopartners/core-components/blob/dev/packages/content-source_story-feed_tag-v4/src/index.test.js

Commit:
test: Add additional checks on tag slug params

matthewroach commented on blocks/story-feed-tag-content-source-block/sources/story-feed-tag.js:
```diff
@@ -1,11 +1,39 @@
+    throw new Error('tagSlug parameter is required');
+  }
+
+  return `/content/v4/search/published?q=taxonomy.tags.slug:${tagSlug}&size=${feedSize}&from=${feedOffset}&sort=display_date:desc&website=${website}`;
```
`feedSize` and `feedOffset` do not have a default, which causes an error. In my PR I added defaults here to account for this

JackHowa replied:
will do thanks

JackHowa replied:
updated @matthewroach

JackHowa replied:
before 

throws error on tag slug but no feedSize or feedoffset

after 

<img width="1383" alt="Screen Shot 2021-07-26 at 10 51 25" src="https://user-images.githubusercontent.com/5950956/127020308-87ffd72e-104d-4c29-8cee-2dc29556b52e.png">

matthewroach reviewed (changes requested):
Would be nice to add defaults for `feedSize` and `feedOffset`

Commit:
fix: Add defaults for feedsize and feed offset per feedback

Via https://github.com/WPMedia/fusion-news-theme-blocks/pull/1004#discussion_r676715925

Commit:
Merge branch 'canary' into TMEDIA-280-story-feed-tag-perf

Commit:
fix: Use default params inline rather than undefined fallback

Commit:
Merge branch 'TMEDIA-280-story-feed-tag-perf' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-280-story-feed-tag-perf
